src/components/setup-sharehouse/SetupStepper.tsx

```diff
@@ -0,0 +1,243 @@
+'use client';
+import { useState } from 'react';
+import { Input } from '@/components/ui/input';
+import { ICategory, ITenant } from '@/types/commons';
+import {
+  Accordion,
+  AccordionItem,
+  AccordionContent,
+} from '@/components/ui/accordion';
+import { AccordionTaskItem } from '@/components/ui/accordion/AccordionTaskItem';
+import { AccordionCategoryItem } from '@/components/ui/accordion/AccordionCategoryItem';
+import { ShareHouseManagementHead } from '@/components/ui/ShareHouseManagementHead';
+import { ScheduleSetting } from './ScheduleSetting';
+import { TenantListItem } from '../ui/tenantList';
+import { DatePicker } from '@/components/ui/datePicker';
+import { SetupConfirmationDrawer } from '@/components/ui/drawers/SetupConfirmationDrawer';
+import { SetupContents } from '@/components/setup-sharehouse/SetupContents';
+import {
+  shareHouseCreationSchema,
+  TShareHouseCreationSchema,
+} from '@/constants/schema';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useForm } from 'react-hook-form';
+import { CONSTRAINTS } from '@/constants/constraints';
+
+const { CATEGORY_MAX_AMOUNT, TENANT_MAX_AMOUNT } = CONSTRAINTS;
+
+interface ISetupStepperProps {
+  initialStep: number;
+  maxSteps: number;
+  tenants: ITenant[];
+  categories: ICategory[];
+}
+
+export const SetupStepper = ({
+  initialStep,
+  maxSteps,
+  tenants,
+  categories,
+}: ISetupStepperProps) => {
+  const formControls = useForm<TShareHouseCreationSchema>({
+    resolver: zodResolver(shareHouseCreationSchema),
+    defaultValues: {
+      name: '',
+      categories,
+      tenants,
+      startDate: new Date().toISOString(),
+      rotationCycle: 7,
+    },
+  });
```

Comment:
Could you add `mode: 'onBlur'` to props for `useForm`?

Currently, error validation is rendered when submitting.
If you add `mode: 'onBlur'`, error validation is rendered when an Input element loses focus.

[About 'mode' of useForm](https://www.react-hook-form.com/api/useform/#mode)

Reply:
Thank you!